tools/run_tests/xds_k8s_test_driver/tests/security_test.py

```diff
@@ -119,38 +122,63 @@ def test_mtls_error(self):
                                    client_tls=True,
                                    client_mtls=False)
 
-        # Create the routing rule map
+        # Create the routing rule map.
         self.td.setup_routing_rule_map_for_grpc(self.server_xds_host,
                                                 self.server_xds_port)
-        # Wait for backends healthy after url map is created
+        # Now that TD setup is complete, Backend Service can be populated
+        # with healthy backends (NEGs).
         self.td.wait_for_backends_healthy_status()
 
-        # Start the client.
+        # Start the client, but don't wait for it to report a healthy channel.
         test_client: _XdsTestClient = self.startSecureTestClient(
             test_server, wait_for_active_server_channel=False)
 
-        # With negative tests we can't be absolutely certain expected
-        # failure state is not caused by something else.
-        # To mitigate for this, we repeat the checks a few times in case
-        # the channel eventually stabilizes and RPCs pass.
-        # TODO(sergiitk): use tenacity retryer, move nums to constants
-        wait_sec = 10
-        checks = 3
-        for check in range(1, checks + 1):
-            self.assertMtlsErrorSetup(test_client)
-            self.assertFailedRpcs(test_client)
-            if check != checks:
-                logger.info(
-                    'Check %s successful, waiting %s sec before the next check',
-                    check, wait_sec)
-                time.sleep(wait_sec)
-
-    @absltest.skip(SKIP_REASON)
+        self.assertClientCannotReachServerRepeatedly(test_client)
+        logger.info(
+            "[SUCCESS] Client's connectivity state is consistent with a mTLS "
+            "error caused by not presenting mTLS certificate to the server.")
+
     def test_server_authz_error(self):
         """Negative test: AuthZ error.
 
         Client does not authorize server because of mismatched SAN name.
+        The order of operations is the same as in `test_mtls_error`.
         """
+        # Create backend service
+        self.td.setup_backend_for_grpc()
+
+        # Start server and attach its NEGs to the backend service, but
+        # until they become healthy.
+        test_server: _XdsTestServer = self.startSecureTestServer()
+        self.setupServerBackends(wait_for_healthy_status=False)
+
+        # Regular TLS setup, but with client policy configured using
+        # intentionality incorrect server_namespace.
+        self.td.setup_server_security(server_namespace=self.server_namespace,
```

Comment:
I traced these calls to see where `server_namespace` is used on the server security side. I noticed it is used in the ECS matcher labels. So presumably the Node metadata also contains a "app: {server_namespace}-{server_name}" entry. This wasn't part of the original test spec (although it might not be a bad idea). Just curious - what purpose does it serve?

Reply:
Good question. The next feature I'm adding is generating a separate namespace for each test run. Since we'll be running tests for different gRPC versions, it's likely at some point there will be a test running at the same time as a test against master. We need to guarantee the uniqueness of the namespace, because ECS are shared by all resources within the same project. Otherwise it's possible that a case where one test created ECS with TLS setting, and another test with mTLS setting - and both tests will fail.